src/adapters/meltdown-1.adapter.ts

```diff
@@ -0,0 +1,43 @@
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking';
+// Use the full path to import instead of `../types`
+import {MeltdownChildren__factory} from '../types/factories/MeltdownChildren__factory';
+
+@injectable(
+  {
+    scope: BindingScope.SINGLETON, // Mark the adapter as a singleton
+  },
+  // Mark it as an extension to staking contracts service
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class MeltdownContractAdapter extends BaseStakingContractAdapter {
+  /**
+   * The contract address
+   */
+  contractAddress = '0xC0c5670D9052D0ee524d7C932382805645Ab3479';
```

Comment:
I just realized that https://polygonscan.com/address/0xc0c5670d9052d0ee524d7c932382805645ab3479#readContract does not have `getStakeInfo`. Please fix.